Python/pytime.c

```diff
@@ -779,19 +779,80 @@ _PyTime_GetMonotonicClockWithInfo(_PyTime_t *tp, _Py_clock_info_t *info)
     return pymonotonic(tp, info, 1);
 }
 
+
+#ifdef MS_WINDOWS
 int
-_PyTime_Init(void)
+_PyTime_GetWinPerfCounterWithInfo(_PyTime_t *t, _Py_clock_info_t *info)
+{
+    static LONGLONG cpu_frequency = 0;
+    static LONGLONG ctrStart;
+    LARGE_INTEGER now;
+    double diff;
+
+    if (cpu_frequency == 0) {
+        LARGE_INTEGER freq;
+        QueryPerformanceCounter(&now);
+        ctrStart = now.QuadPart;
+        if (!QueryPerformanceFrequency(&freq) || freq.QuadPart == 0) {
+            PyErr_SetFromWindowsErr(0);
+            return -1;
+        }
+        cpu_frequency = freq.QuadPart;
+    }
+    QueryPerformanceCounter(&now);
+    diff = (double)(now.QuadPart - ctrStart);
+    if (info) {
+        info->implementation = "QueryPerformanceCounter()";
+        info->resolution = 1.0 / (double)cpu_frequency;
+        info->monotonic = 1;
+        info->adjustable = 0;
+    }
+
+    diff = diff / (double)cpu_frequency;
+    return _PyTime_FromDouble(t, diff, _PyTime_ROUND_FLOOR, SEC_TO_NS);
+}
+#endif
+
+
+int
+_PyTime_GetPerfCounterWithInfo(_PyTime_t *t, _Py_clock_info_t *info)
+{
+#ifdef MS_WINDOWS
+    return _PyTime_GetWinPerfCounterWithInfo(t, info);
```

Comment:
And the call to `_PyTime_FromDouble` was moved into this wrapper function?

It wouldn't be ideal, since the #ifdef logic would exist in two places, but maybe that duplication could be handled via a common macro definition that was used in both places, rather than by incurring the redundant conversion at runtime on Windows.